contracts/utils/RSAPSS.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.16;
+
+error MaskTooLengthy();
+
+library RSAPSS {
+    uint256 constant HASH_LEN = 32;
+    uint256 constant SALT_LEN = 32;
+    uint256 constant MS_BITS = 4095 & 0x7;
+
+    uint256 constant MS_BYTES = 512;
+
+    function verify(
+        bytes memory message_,
+        bytes memory s_,
+        bytes memory e_,
+        bytes memory n_
+    ) internal view returns (bool) {
+        if (s_.length == 0 || e_.length == 0 || n_.length == 0) {
+            return false;
+        }
+
+        bytes memory decipher_ = decrypt(s_, e_, n_);
+
+        return pss(message_, decipher_);
+    }
+
+    function decrypt(
```

Comment:
Why not use existing `RSA` lib for that?

Reply:
moved to rsa, in this case should we use the proper function naming?
As "decrypt" and "verify" functions don't give me the real understanding which function is supposed to be for RSA and RSAPSS